sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -439,8 +447,9 @@ case class TransformWithStateExec(
       new Path(getStateInfo.checkpointLocation,
         s"${getStateInfo.operatorId.toString}")
 
-    val storeNamePath = new Path(stateCheckpointPath, storeName)
-    new Path(new Path(storeNamePath, "_metadata"), "schema")
+    val stateSchemaPath = new Path(stateCheckpointPath, "_stateSchema")
```

Comment:
Yes correct - newer operators would follow a new structure where:
- within the operatorId dir, there are 2 subdirs:
 - for the operator metadata, we have the `_metadata` dir
 - for the state schema, we have the `_stateSchema` dir

Reply:
Shall we describe the directory structure for previous version vs proposal, in the PR description, so that we wouldn't require to look into code? Ideally, we should have some doc about checkpoint structure (now the complication is something we can't expect someone to understand without documentation) but I agree this is a general issue and we want to solve this in general, not from this PR.

Reply:
OK done - lmk if you think more detail is needed ?